Framework/PythonInterface/plugins/algorithms/CreatePoleFigureTableWorkspace.py

```diff
@@ -0,0 +1,231 @@
+# import mantid algorithms, numpy and matplotlib
+import numpy as np
+from mantid.kernel import V3D
+from mantid.api import AlgorithmFactory, MatrixWorkspaceProperty, ITableWorkspaceProperty, PythonAlgorithm, PropertyMode
+from mantid.kernel import Direction, FloatBoundedValidator
+from mantid.geometry import ReflectionGenerator
+
+
+class CreatePoleFigureTableWorkspace(PythonAlgorithm):
+    def category(self):
+        return "Diffraction\\Engineering"
+
+    def name(self):
+        return "CreatePoleFigureTableWorkspace"
+
+    def summary(self):
+        return "Creates a table storing the pole figure data extracted from the input workspaces."
+
+    def version(self):
+        return 1
+
+    def PyInit(self):
+        self.declareProperty(
+            MatrixWorkspaceProperty("InputWorkspace", defaultValue="", direction=Direction.Input),
+            doc="The workspace containing the input data.",
+        )
+        self.declareProperty(
+            ITableWorkspaceProperty("PeakParameterWorkspace", "", Direction.Input, PropertyMode.Optional),
+            doc="Optional workspace containing the fitted peak data.",
+        )
+        self.declareProperty(
+            ITableWorkspaceProperty("OutputWorkspace", "", Direction.Output),
+            doc="Output workspace containing the table of alphas, betas and intensities.",
+        )
+        self.declareProperty(
+            name="Reflection",
```

Comment:
Is it possible to define this with `PropertyMode.Optional` (or set default to 0,0,0) with data type `FloatArrayProperty` such that the function `_parse_hkl` wouldn't be needed?